Gems/Atom/RHI/Metal/Code/Include/Atom/RHI.Reflect/Metal/BufferPoolDescriptor.h

```diff
@@ -18,6 +18,7 @@ namespace AZ
             : public RHI::BufferPoolDescriptor
         {
         public:
+            AZ_CLASS_ALLOCATOR(BufferPoolDescriptor, SystemAllocator, 0);
```

Comment:
Some of these calls are missing the AZ::SystemAllocator.
I know they are in the AZ namespace, but other calls added the "AZ" as well.

Reply:
To make sure I was using the same allocator as the parent class, I used a macro in vim to add the `AZ_CLASS_ALLOCATOR` line. The macro did this:
* Go to the subclass's base class's definition
* Search for its `AZ_CLASS_ALLOCATOR` line, and copy it
* Go back to the subclass, paste the `AZ_CLASS_ALLOCATOR` line from the base class
* Replace the base class's name in the new macro with the subclass's name

The result of this is that if the base class spelled the allocator as `SystemAllocator` instead of `AZ::SystemAllocator`, then that's how it is spelled in the subclass's use as well. For this specific instance, the base class goes through `BufferPoolDescriptor` to `ResourcePoolDescriptor`:
https://github.com/o3de/o3de/blob/1c914f819c473ef4492360b02424e531727e846b/Gems/Atom/RHI/Code/Include/Atom/RHI.Reflect/ResourcePoolDescriptor.h#L24

So it is consistent in that it's spelled the same as the base class, rather than consistent in that a given allocator is always spelled the same way.